docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,374 @@
+
+For more details about Azure AD Pod Identity please visit this [link](https://github.com/Azure/aad-pod-identity)  
+
+## <a name='Motivation'></a>Motivation
+
+AZWI gives the capability to federate the credentials with external identity providers in a Kubernetes native way. This approach overcomes several limitations of Azure AD Identity as mentioned below.
+- Removes scale and performance issues that existed for identity assignment.
+- Supports K8s clusters hosted in any cloud or on premise.
+- Supports both Linux and Windows workloads and removes the need for CRD and pods that intercept IMDS traffic.
+
+From CAPZ perspective
+- The AAD pod identity pod has to be deployed on all the nodes where the capz pod can be potentially scheduled.
+- It has a dependency with the CNI and requires modifying iptables on the node.
+- AAD pod identity is now deprecated. 
+
+To learn more about AZWI please visit this link https://azure.github.io/azure-workload-identity/docs/introduction.html
+
+### <a name='Goals'></a>Goals
+
+- Enable CAPZ pod to use AZWI on the management cluster to authenticate to Azure to create/update/delete resources as part of workload cluster lifecycle management
+
+## <a name='Proposal'></a>Proposal
+
+In AZWI model, Kubernetes cluster itself becomes the token issuer issuing tokens to Kubernetes service accounts. These service accounts can be configured to be trusted on Azure AD application or user assigned managed identity. Workloads/pods can use this service account token which is projected to it's volume and can exchange the projected service account token for an Azure AD token.
+
+The first step for creating a Kubernetes cluster using capz is creating a bootstrap cluster and then deploying capi, capz and other helper components. On a high level the workflow looks the following to be able to use AZWI and create a management cluster.
+
+**Bootstrap Cluster**
+
+- The operator/admin generates signing key pair or BYO key pair. 
+
+- The bootstrap cluster is configured with appropriate flags on kube-apiserver and kube-controller-manager and the key pairs are mounted on the container path for control plane node. See [this](#SetServiceAccountSigningFlags) section for more details on this.
+	- kube-apiserver flags
+		- --service-account-issuer
+		- --service-account-signing-key-file
+		- --service-account-key-file
+	- kube-controller-manager
+		- --service-account-private-key-file
```

Comment:
Have you considered to provide a secret with clustername-sa before creating the cluster, so then KCP will automatically use it (see https://cluster-api.sigs.k8s.io/tasks/certs/using-custom-certificates.html)

This will have also the nice side effect that also move should take care of this secret without any additional operation

Reply:
+1 Ideally we should let KCP pick up pre-created service account key-pair

Reply:
@sonasingh46 thoughts on this?

Reply:
Yes. Have included this in the workflow.

Reply:
But for moving a existing cluster to use workload identity -- that will force to use the same key pairs encapsulated in the <cluster-name>-sa and we can think of removing this dependency going forward.